BlocNotes/NoteEntryListViewController.m

```diff
@@ -0,0 +1,181 @@
+    return [sectionInfo numberOfObjects];;
+}
+
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
+    static NSString *cellIdentifier = @"cell";
+    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:cellIdentifier forIndexPath:indexPath];
+    
+    NoteEntry *noteEntry = [self.fetchedResultsController objectAtIndexPath:indexPath];
+    
+    cell.textLabel.text = noteEntry.body;
+    
+    return cell;
+}
+
+- (NSFetchRequest *) entryListFetchRequest {
+    NSFetchRequest *fetchRequest = [NSFetchRequest fetchRequestWithEntityName:@"NoteEntry"];
+    fetchRequest.sortDescriptors = @[[NSSortDescriptor sortDescriptorWithKey:@"date" ascending:NO]];
+    
+    return fetchRequest;
+}
+
+- (NSFetchedResultsController *) fetchedResultsController {
+    if (_fetchedResultsController !=nil) {
+        return _fetchedResultsController;
+    }
+    
+    CoreDataStack *coreDataStack = [CoreDataStack defaultStack];
+    NSFetchRequest *fetchRequest = [self entryListFetchRequest];
+    
+    _fetchedResultsController = [[NSFetchedResultsController alloc] initWithFetchRequest:fetchRequest managedObjectContext:coreDataStack.managedObjectContext sectionNameKeyPath:@"sectionName" cacheName:nil];
+    _fetchedResultsController.delegate = self;
+    
+    return _fetchedResultsController;
+}
+
+- (UITableViewCellEditingStyle)tableView:(UITableView *)tableView editingStyleForRowAtIndexPath:(NSIndexPath *)indexPath {
+    return UITableViewCellEditingStyleDelete;
+}
+
+- (void) tableView:(UITableView *)tableView commitEditingStyle:(UITableViewCellEditingStyle)editingStyle forRowAtIndexPath:(nonnull NSIndexPath *)indexPath {
+    NoteEntry *note = [self.fetchedResultsController objectAtIndexPath: indexPath];
+    CoreDataStack *coreDataStack = [CoreDataStack defaultStack];
+    [[coreDataStack managedObjectContext] deleteObject:note];
+    [coreDataStack saveContext];
+}
+
+- (void) addWasPressed {
+    NewEntryViewController *newEntry = [[NewEntryViewController alloc] init];
+
+    UINavigationController *navigationController = [[UINavigationController alloc] initWithRootViewController:newEntry];
+    [self presentViewController:navigationController animated:YES completion: nil];
+}
+
+- (void) controllerWillChangeContent:(NSFetchedResultsController *)controller {
+    [self.tableView beginUpdates];
+}
+
+- (void) tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
+    NewEntryViewController *newEntry = [[NewEntryViewController alloc] init];
+    
+    UINavigationController *navigationController = [[UINavigationController alloc] initWithRootViewController:newEntry];
+    [self presentViewController:navigationController animated:YES completion: nil];
+    
+    newEntry.noteEntry = [self.fetchedResultsController objectAtIndexPath:indexPath];
+}  // What's efficient way to do this?
```

Comment:
Not sure what this question is specifically referring to... Are you asking about how to present the NewEntryViewController more efficiently? As far as that goes, this looks fine to me.